<a href="https://colab.research.google.com/github/izzat-ai/learning-ai/blob/main/projects/formula_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

- **Juzeppe Farinaning 1950-yilgi chempionligidan bugungi kungacha bo'lgan har bir poyga, haydovchi, konstruktor, saralash sessiyasi va F1 tarixidagi o'rni**

- `cons_standing` - Ochkolar va pozitsiyalar
- `cons` - Jamoalar nomlari
- `races` - Yillar va poyga nomlari
- `results` - Har bir poyga natijalari


In [1]:
import numpy as np
import pandas as pd

In [2]:
cons_standing = pd.read_csv("/content/drive/MyDrive/AI learning/ai_project_2026/formula1/f1_constructor_standings.csv")
cons = pd.read_csv("/content/drive/MyDrive/AI learning/ai_project_2026/formula1/f1_constructors.csv")
races = pd.read_csv("/content/drive/MyDrive/AI learning/ai_project_2026/formula1/f1_races.csv")
results = pd.read_csv("/content/drive/MyDrive/AI learning/ai_project_2026/formula1/f1_results.csv")

#####**Data cleaning**

1. cons_standing cleaning
2. cons cleaning
3. races cleaning
4. results cleaning

In [3]:
cons_standing.head()

,season,position,points,wins,constructor_id,constructor
0,1958,1.0,48.0,6,vanwall,Vanwall
1,1958,2.0,40.0,2,ferrari,Ferrari
2,1958,3.0,31.0,2,cooper,Cooper
3,1958,4.0,18.0,0,brm,BRM
4,1958,5.0,6.0,0,maserati,Maserati


In [4]:
cons_standing.shape

(924, 6)

In [12]:
# check
def check_health(df, name):
    print(f"--- {name} data report ---")
    print(f"null values:\n{df.isnull().sum()}\n")
    print(f"number of duplicates: {df.duplicated().sum()}")
    print("-" * 30)

check_health(cons, "Constructors")
check_health(cons_standing, "Standings")
check_health(races, "Races")

--- Constructors data report ---
null values:
constructor_id    0
name              0
nationality       0
url               0
dtype: int64

number of duplicates: 0
------------------------------
--- Standings data report ---
null values:
season              0
position          212
points              0
wins                0
constructor_id      0
constructor         0
dtype: int64

number of duplicates: 0
------------------------------
--- Races data report ---
null values:
season          0
round           0
race_name       0
circuit_id      0
date            0
time          731
url             0
dtype: int64

number of duplicates: 0
------------------------------


- `const` dataframe - No empty (null) values ​​and no duplicates
- `cons_standing` dataframe - The position column contains 212 null values . In F1 history, some constructors may have been disqualified or not finished the season
- `races` dataframe - 731 empty values ​​(null) . The exact hours and minutes of older races (e.g., from the 1950s-60s) may not be stored in the historical database